main.py

```diff
@@ -0,0 +1,153 @@
+#########################################################
+
+import pygame
+import sys
+import time
+
+from snakes_battle.board import Board
+from snakes_battle.snakes_ai.random_snake import RandomSnake
+from snakes_battle.snakes_ai.simple_snake import SimpleSnake
+from snakes_battle.snakes_ai.manual_control_snake import ManualSnake
+from snakes_battle.snakes_ai.manual_control_snake_wasd import ManualSnakeWASD
+from snakes_battle import logic
+from snakes_battle.graphics import GameGraphics
+from tkinter import messagebox,Tk
+import settings
+from snakes_battle.fruit import FruitKind, Fruit
+
+ai_classes_available = [
+    { "class": RandomSnake, "should_play": False },
+    { "class": SimpleSnake, "should_play": False },
+    { "class": ManualSnake, "should_play": False },
+    { "class": ManualSnakeWASD, "should_play": False }
+    ]
+
+def main():
+    should_exit = False
+    game_running = False
+    game_menus = True
+
+    graphics = GameGraphics(ai_classes_available)
+
+    while (should_exit == False):
+        if (game_running == True):
+            playing_classes = [x['class'] for x in ai_classes_available if x["should_play"] == True]
+            if (len(playing_classes) > 0):
+                run_game(playing_classes)
+            else:
+                root = Tk()
+                root.withdraw()
+                messagebox.showinfo("Error", "Cannot play without any snakes chosen!")
+                root.destroy()
+            game_menus = True
+            game_running = False
+
+        elif (game_menus == True):
+            menus_return = run_menus(graphics)
+            if (menus_return['action'] == "Exit"):
+                pygame.quit()
+                should_exit = True
+            elif(menus_return['action'] == "Play"):
+                game_menus = False
+                game_running = True
+
+
+
+def run_menus(graphics):
+    global ai_classes_available # A global just for it to be more accessible
+
+    menu_running = True
+
+    main_menu = True
+    snake_picker = False
+
+    while (menu_running == True):
+        events = pygame.event.get()
+        for event in events:
+            if event.type == pygame.QUIT:
+                return "Exit"
+            if event.type == pygame.KEYDOWN:
+                for index, class_dict in enumerate(ai_classes_available):
+                    if (event.key == settings.PYGAME_START_NUMBER_PRESS_VALUE + index):
+                        class_dict['should_play'] = not class_dict['should_play']
+
+
+        if (main_menu == True):
+            menu_action = graphics.draw_menu(ai_classes_available, events)
+            if (menu_action == 'Exit'):
+                return {"action": menu_action, "args": ""}
+            elif (menu_action == 'Play'):
+                return {"action": menu_action, "args": ""}
+
+        elif (snake_picker == True):
+            pass
+
+def run_game(playing_classes):
+    graphics = GameGraphics(ai_classes_available)
```

Comment:
I would have pass the graphics object from main function to this function.

Reply:
This is a bug fix.
It started as you say - but for some reason it caused the game after a few runs from the menu to crash.
That fixed it - I guess something in the graphics class is not optimized - but for now that worked well.